R/select_rows.r

```diff
@@ -17,12 +20,14 @@
 #' select_rows(iris, c(TRUE,FALSE)) # Select only odd rows
 #' iris2 <- iris; rownames(iris2) <- paste0("row", 1:nrow(iris2))
 #' select_rows(iris, c("row10", "row51")) # Select rows by name
-select_rows <- function(dataframe, rows, whole = FALSE, ...) {
+select_rows <- function(dataframe, rows, whole = FALSE, cols_to_check = NULL, ...) {
   force(rows); force(whole); args <- force(list(...))
+  cols <- names(dataframe)
+  if (!is.null(cols_to_check)) cols <- intersect(cols, cols_to_check)
   eval(substitute({
     dataframe <- dataframe[if(is.function(rows)) {
-      if (whole) rows(dataframe) else apply(dataframe, 1, rows)
-    } else rows, ]
+      if (whole) rows(dataframe[, cols, drop = FALSE]) else apply(dataframe[, cols, drop = FALSE], 1, rows)
+    } else rows, , drop = FALSE]
```

Comment:
Don't touch it please -- no need to modify the closure environment upstairs.

On Wed, Mar 23, 2016 at 1:45 PM, David <notifications@github.com> wrote:

> In R/select_rows.r
> <https://github.com/robertzk/syberiaMungebits/pull/55#discussion_r57215282>
> :
>
> >    eval(substitute({
> >      dataframe <- dataframe[if(is.function(rows)) {
> > -      if (whole) rows(dataframe) else apply(dataframe, 1, rows)
> > -    } else rows, ]
> > +      if (whole) rows(dataframe[, cols, drop = FALSE]) else apply(dataframe[, cols, drop = FALSE], 1, rows)
> > +    } else rows, , drop = FALSE]
>
> yeah... I really wanted to rewrite this ugly mess but didn't want the risk
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/robertzk/syberiaMungebits/pull/55/files/bf66df529386e5588d00626ce4c6b415131cb9b0#r57215282>
>



-- 
-Robert Krzyzanowski